source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,206 @@
+Ian Ray, 25 July 2023.
+
+Formalizing the auxilary notion of a delta-V-complete poset which is
+sufficient for the main theorems of *link Tom de Jong's thesis*
+involving impredicativity (in the form of resizing axioms) and order theory.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import MLTT.Two-Properties
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Logic
+
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+
+module Posets.Delta-V-CompletePoset
+ (pt : propositional-truncations-exist)
+ (fe : Fun-Ext)
+  where
+
+open import Locales.Frame pt fe
+
+open import Posets.2ElementPoset pt fe
+
+module δ_𝓥-complete-poset {𝓤 𝓦 : Universe} (𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+ δ : (x y : ∣ A ∣ₚ) → (P : Ω 𝓥) → (𝟙{𝓥} + P holds) → ∣ A ∣ₚ 
+ δ x y P (inl _) = x
+ δ x y P (inr _) = y
+
+ open Joins (rel-syntax A)
+
+ is-δ_𝓥-complete : 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+ is-δ_𝓥-complete = (x y : ∣ A ∣ₚ) → (x ≤[ A ] y) holds → (P : Ω 𝓥) →
+   Σ s ꞉ ∣ A ∣ₚ , (s is-lub-of ((𝟙 + P holds) , (δ x y P))) holds
+
+\end{code}
+
+Future work: I would be nice to show that classically every poset is
+δ_𝓥 complete and to provide the critical examples of δ_𝓥 complete posets
+such as 𝓥-sup lattices, etc. We should also show that is-δ_𝓥-complete is
+a subsingleton.
+
+We now define the type of δ_𝓥 complete posets.
+
+\begin{code}
+
+δ_𝓥-complete-Poset : (𝓤 𝓦 𝓥 : Universe) → (𝓤 ⁺) ⊔ (𝓦 ⁺) ⊔ (𝓥 ⁺) ̇
+δ_𝓥-complete-Poset 𝓤 𝓦 𝓥 = Σ A ꞉ Poset 𝓤 𝓦 , δ_𝓥-complete-poset.is-δ_𝓥-complete 𝓥 A
+
+Poset-δ : (D : δ_𝓥-complete-Poset 𝓤 𝓦 𝓥) → Poset 𝓤 𝓦
+Poset-δ D = pr₁ D
+
+is-δ_𝓥-complete-δ : (D : δ_𝓥-complete-Poset 𝓤 𝓦 𝓥) →
+  δ_𝓥-complete-poset.is-δ_𝓥-complete 𝓥 (Poset-δ D)
+is-δ_𝓥-complete-δ D = pr₂ D
+
+\end{code}
+
+\begin{code}
+
+module non-trivial-posets {𝓤  𝓦  𝓥 : Universe} (A : Poset 𝓤 𝓦) where
+ is-non-trivial-poset : 𝓤 ⊔ 𝓦 ̇
+ is-non-trivial-poset =  Σ x ꞉ ∣ A ∣ₚ ,
+  ( Σ y ꞉ ∣ A ∣ₚ , (x ≤[ A ] y) holds × ¬ (x ＝ y))
+
+ lower : is-non-trivial-poset → ∣ A ∣ₚ
+ lower i = pr₁ i
+
+ upper : is-non-trivial-poset → ∣ A ∣ₚ
+ upper i = pr₁ (pr₂ i)
+
+ order : (i : is-non-trivial-poset) → (lower i ≤[ A ] upper i) holds
+ order i = pr₁ (pr₂ (pr₂ i))
+
+ nequal : (i : is-non-trivial-poset) → ¬ (lower i ＝ upper i)
+ nequal i = pr₂ (pr₂ (pr₂ i))
+
+ open Joins (rel-syntax A)
+ open δ_𝓥-complete-poset 𝓥 A
+
+ wlem-lemma : (i : is-non-trivial-poset) → (P : Ω 𝓥) →
```

Comment:
You could try something like:
```
 module _ (𝓥 : Universe)
          (i : is-non-trivial-poset)
        where

  open Joins (rel-syntax A)
  open δ_𝓥-complete-poset 𝓥 A
  private
   x = lower i
   y = upper i

  wlem-lemma : (P : Ω 𝓥) →
   ((x is-lub-of ((𝟙 + P holds) , δ x y P)) holds → ¬ (P holds)) ×
   ((y is-lub-of ((𝟙 + P holds) , δ x y P)) holds → ¬ ¬ (P holds))
  pr₁ (wlem-lemma P) r p = nequal i (≤-is-antisymmetric A (order i) (pr₁ r (inr p)))
  pr₂ (wlem-lemma P) r np = nequal i (≤-is-antisymmetric A (order i) (pr₂ r ((x , h))))
   where
    h : (x is-an-upper-bound-of ((𝟙 + P holds) , δ x y P)) holds
    h (inl ✯) = ≤-is-reflexive A x
    h (inr p) = 𝟘-induction (np p)
```
to make the whole thing a bit easier on the eyes.

Reply:
This is really nice! Thanks for the suggestion of making x and y private variables.